### PR TITLE
Improve error handling and logging

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #350 

## Description
This PR improves how we handle errors and more importantly not log errors that are user errors (invalid inputs).

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels 

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive error handling mechanism in various components, improving clarity and maintainability.
	- Added specific error types for better feedback on transaction issues, such as `FailedTransaction` and `InvalidTransaction`.
  
- **Bug Fixes**
	- Enhanced error validation in transaction arguments to provide more specific error messages.

- **Chores**
	- Updated import paths for error handling to consolidate definitions under a `models` package, improving code organization.

- **Tests**
	- Refactored test cases to utilize new error definitions, enhancing consistency in error handling within tests.
	- Modified error handling in `blockStorage` and `receiptStorage` functions in tests for better clarity.

### CODE REVIEW

File: services/logs/filter_test.go [hunk 1]
```diff
@@ -9,10 +9,11 @@ import (
 	gethTypes "github.com/onflow/go-ethereum/core/types"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"github.com/syndtr/goleveldb/leveldb/errors"
```
Comment (m-Peter) [inline services/logs/filter_test.go:12]: Where did that come from? :thinking: Probably wrong import?

Comment (sideninja) [reply services/logs/filter_test.go:12]: yep

Comment (sideninja) [reply services/logs/filter_test.go:12]: tnx

**Review by m-Peter — APPROVED** [review]
LGTM! :100:

**Pull request merged**